core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -248,9 +248,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
       val logInfos: Seq[FileStatus] = statusList
         .filter { entry =>
           try {
-            getModificationTime(entry).map { time =>
-              time >= lastScanTime
-            }.getOrElse(false)
+            entry.getModificationTime() >= lastScanTime
```

Comment:
That's a great point, thank you for bringing that up.
I wasn't sure if filtering out directory here was too early, and instead should pass it on and let `EventLoggingListener.openEventLog` decide if it was valid (directory would not be).
For now I add the check.

Also perhaps we should pass a Filter to `fs.listStatus` if the file naming convention is known (again assuming we want `checkForLogs` as the gatekeeper as above).

And I'm pretty sure it no longer throws AcessControlException on [line 253](https://github.com/apache/spark/pull/10860/files#diff-a7befb99e7bd7e3ab5c46c2568aa5b3eR253) since `listStatus` is not getting called in that path.